### ISSUE TITLE
Error: cannot find symbol method use(Factory)

### ISSUE THREAD

**Issue by hardartcore** [segmentio/analytics-android#452]
Using the latest library in out application we are getting this error while trying to build our app:

`Error:(106, 17) error: cannot find symbol method use(Factory)`

Segment libraries and integrations which we are using are:
`compile 'com.segment.analytics.android:analytics:4.1.4'`

Localytics integration:
`compile('com.segment.analytics.android.integrations:localytics:1.0.2') {transitive = true}`

AppsFlyer integration:
`compile('com.segment.analytics.android:analytics-integration-appsflyer:3.4.0') {transitive = true;}`

And here is the initialization in our main Application's class `onCreate` :

```
Analytics analytics = new Analytics.Builder(this, Constants.SEGMENT_WRITE_KEY)
                .use(LocalyticsIntegration.FACTORY)
                .logLevel(BuildConfig.DEBUG ? Analytics.LogLevel.DEBUG : Analytics.LogLevel.NONE)
                .build();
Analytics.setSingletonInstance(analytics);
```

And the error which Android Studio shows is: 
`Error:(106, 17) error: cannot find symbol method use(Factory)`

I've tried everything which I can think of, but no luck.
Any advice how can we fix this issue?

**Comment by f2prateek** [comment]
What's the output of `./gradlew dependencies`? Did you also try going a clean build?

what if you try ` .use(new Factory{} () {....} )` and fill it in using a stubbed implementation - does that still compile? 

Also for future reference, the best place to ask these questions is emailing friends@segment.com.

**Comment by hardartcore** [comment]
Nope that doesn't compile either.
So after a few hours digging and trying different stuff I find out the solution.
In your Localytics integration library you are using 4.0.0 version of your main Segment library which is not the latest one which I am using in our app. Same story in AppsFlyer integration library.

So it would be really good after releasing a new version of your main library to update other integrations libraries to use the latest one, because it confuses proguard too, because you have duplicate versions of same class.

So my fix for this issue is:
```
    vcompile 'com.segment.analytics.android:analytics:4.1.4'
    compile('com.segment.analytics.android.integrations:localytics:1.0.2') {
        exclude group: 'com.segment.analytics.android', module: 'analytics'
    }
    compile 'com.appsflyer:af-android-sdk:4.4.0@aar'
```

**Comment by f2prateek** [comment]
That's not the issue. I have a working example with the following configuration.

```
compile 'com.segment.analytics.android:analytics:4.1.4'
compile('com.segment.analytics.android.integrations:localytics:1.0.2') {
  transitive = true
}
```

And if you check the `./gradlew dependencies` output for this, you'd see:

```
compile - Classpath for compiling the main sources.
+--- com.segment.analytics.android:analytics:4.1.4 -> project :analytics
+--- com.segment.analytics.android.integrations:localytics:1.0.2
|    +--- com.android.support:support-v4:23.1.1
|    |    \--- com.android.support:support-annotations:23.1.1
|    \--- com.localytics.android:library:3.8.1
```

The integrations don't depend on a specific version when you build your app. They mark the dependency as `provided` (e.g. https://github.com/segment-integrations/analytics-android-integration-localytics/blob/master/build.gradle#L47.) so whatever you are compiling gets used - rather than pulling in a version themselves (as you see in the dependencies output).

The issue is probably coming from a custom build setup (such as proguard) in your app.

And lastly, the `use` method is available since 4.0.0 - so even *if* the libraries were pulling in 4.0.0 - the `use` method should be available.
